testing/ras/test_main_c.cpp

```diff
@@ -0,0 +1,484 @@
+ *             page_fault_errors returns FPGA_INVALID_PARAM.<br>
+ */
+TEST(ras_c, invalid_page_fault_errors){
+  EXPECT_EQ(FPGA_INVALID_PARAM, page_fault_errors());
+}
+
+/**
+ * @test       main_params_01
+ * @brief      Test: ras_main
+ * @details    When ras_main is called with an valid command,<br>
+ *             that includes errors injection, ras_main returns zero.<br>
+ */
+TEST_P(ras_c_p, main_params_01){
+  char zero[20];
+  char one[20];
+  char two[20];
+  char three[20];
+  char four[20];
+  char five[20];
+  char six[20];
+  char seven[20];
+  char eight[20];
+  char nine[20];
+  char ten[20];
+  char eleven[20];
+  char twelve[20];
+  char thirteen[20];
+  char fourteen[20];
+  char fifteen[20];
+  char sixteen[20];
+
+  strcpy(zero, "ras");
+  strcpy(one, "--segment");
+  sprintf(two, "%d", platform_.devices[0].segment);
+  strcpy(three, "-B");
+  sprintf(four, "%d", platform_.devices[0].bus);
+  strcpy(five, "-D");
+  sprintf(six, "%d", platform_.devices[0].device);
+  strcpy(seven, "-F");
+  sprintf(eight, "%d", platform_.devices[0].function);
+  strcpy(nine, "-S");
+  sprintf(ten, "%d", platform_.devices[0].socket_id);
+  strcpy(eleven, "-P");
+  strcpy(twelve, "-Q");
+  strcpy(thirteen, "-R");
+  strcpy(fourteen, "-O");
+  strcpy(fifteen, "-N");
+  strcpy(sixteen, "-C");
+
+  char *argv[] = { zero, one, two, three, four,
+                   five, six, seven, eight, nine,
+                   ten, eleven, twelve, thirteen,
+                   fourteen, fifteen, sixteen };
+
+  EXPECT_EQ(ras_main(17, argv), 0);
+}
+
+/**
+ * @test       main_params_05
+ * @brief      Test: ras_main
+ * @details    When ras_main is called with an valid command option,<br>
+ *             injecting mmio error, ras_main returns zero.<br>
+ */
+TEST_P(ras_c_p, main_params_05){
+  char zero[20];
+  char one[20];
+  char two[20];
+  char three[20];
+  char four[20];
+  char five[20];
+  char six[20];
+  char seven[20];
+  char eight[20];
+
+  strcpy(zero, "ras");
+  strcpy(one, "-B");
+  sprintf(two, "%d", platform_.devices[0].bus);
+  strcpy(three, "-P");
+  strcpy(four, "-E");
+  strcpy(five, "-G");
+  strcpy(six, "-H");
+  strcpy(seven, "-I");
+  strcpy(eight, "-C");
+
+  char *argv[] = { zero, one, two, three, four,
+                   five, six, seven, eight };
+
+  EXPECT_EQ(ras_main(9, argv), 0);
+}
+
+/**
+ * @test       invalid_cmd_01
+ * @brief      Test: ras_main
+ * @details    When ras_main is called with only 1 arg,<br>
+ *             main returns 1.<br>
+ */
+TEST_P(ras_c_p, invalid_cmd_01){
+  char zero[20];
+  strcpy(zero, "ras");
+
+  char *argv[] = { zero }; 
+  EXPECT_EQ(ras_main(1, argv), 1);
+}
+
+/**
+ * @test       invalid_cmd_02
+ * @brief      Test: ras_main
+ * @details    When ras_main is called with an invalid command option,<br>
+ *             main returns 2.<br>
+ */
+TEST_P(ras_c_p, invalid_cmd_02){
+  char zero[20];
+  char one[20];
+  
+  strcpy(zero, "ras");
+  strcpy(one, "-h");
```

Comment:
-h is technically a valid option, since it appears in GETOPT_STRING.  Pass in something that doesn't appear there, like -Y.